src/controllers/search.controller.js

```diff
@@ -0,0 +1,39 @@
+import SearchService from '../services/search.service';
+import ResponseService from '../services/response.service';
+import { paginationHelper } from '../helpers';
+
+/**
+ *
+ *
+ * @class searchController
+ */
+class searchController {
+  /**
+   *
+   *
+   * @static
+   * @param {req} req
+   * @param {res} res
+   * @returns {response} @memberof searchController
+   */
+  static async searchRequests(req, res) {
+    const { term, page, limit } = req.query;
+    const { userData } = req;
+    const offset = (page - 1) * limit;
+
+    const pagination = {
+      limit,
+      offset: Number.isNaN(offset) ? undefined : offset
```

Comment:
Offset should be zero by default. Also, line 22 cannot return a NaN value so the check here is unnecessary.

Reply:
@adesege Because pagination in this search is optional, I was thinking if the user searches without providing the page value or limit value, he should get the data without pagination. In that situation, line 22 would return NaN and that's why I was checking first.  If you say that offset should be zero by default, there would be no pagination. Can you give me your suggestions? because for me without setting the default values for page and limit which would be pagination, I'm not finding any other option except checking.